sprockets/http/app.py

```diff
@@ -120,6 +130,10 @@ def stop(self, io_loop):
         for callback in self.on_shutdown_callbacks:
             try:
                 maybe_future = callback(self.tornado_application)
+
+                if ASYNCIO_TORNADO and asyncio.iscoroutine(maybe_future):
+                        maybe_future = asyncio.create_task(maybe_future)
+
                 if concurrent.is_future(maybe_future):
```

Comment:
Sorry, I am not well versed in Tornado's concurrent module. Please tell me if I understand this right.

Following this code...

- Suppose that ASYNCIO_TORNADO is True, and maybe_future was a coroutine on line 134.
- In Tornado's "concurrent.py",
- - FUTURES includes concurrent.future.Futures and asyncio.Future
- - The is_future() method checks if its argument is an instance of FUTURES.
- Therefore if maybe_future is an instance of asyncio.Future, the condition on line 137 here will evaluate to true, and we will enter the code on line 139 that sets running_async to True.